agent/api/ecsclient/client.go

```diff
@@ -259,10 +259,37 @@ func getCpuAndMemory() (int64, int64) {
 }
 
 func (client *APIECSClient) getAdditionalAttributes() []*ecs.Attribute {
-	return []*ecs.Attribute{&ecs.Attribute{
+	attributes := []*ecs.Attribute{&ecs.Attribute{
 		Name:  aws.String("ecs.os-type"),
 		Value: aws.String(api.OSType),
 	}}
+	mac, err := client.ec2metadata.PrimaryENIMAC()
+	if err != nil {
+		seelog.Warnf("Unable to get the MAC address of the primary network interface: %v", err)
+		return attributes
+	}
+
+	vpcID, err := client.ec2metadata.VPCID(mac)
+	if err != nil {
+		seelog.Warnf("Unable to get the VPC ID of the primary network interface: %v", err)
+		return attributes
```

Comment:
What's the rationale behind immediately returning upon the first encountered error? It's conceivable (though unlikely) that retrieval of the VPCID fails but PrimaryENIMAC and SubnetID are successful. If this happens, our return value contains neither of the successfully retrieved attributes.